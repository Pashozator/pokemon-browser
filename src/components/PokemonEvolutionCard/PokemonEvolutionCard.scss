.pokemon-evolution {
  &:not(:last-child) {
    margin-right: 10px;
  }

  .pokemon-evolution-card {
    padding: 10px;
    align-items: center;
    justify-content: flex-start;
    flex-direction: row;
    display: flex;

    &:hover {
      cursor: pointer;
      box-shadow: 0 3px 4px 0 rgba(0, 0, 0, .3);
      transition: all ease .3s;
    }

    img {
      margin-right: 10px;
      max-height: 45px;
    }
  }
}
