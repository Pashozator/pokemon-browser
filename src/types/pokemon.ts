export interface Pokemon {
	id: string;
	number: number;
	name: string;
	image: string;
	maxHP: number;
	fleeRate: number;
	maxCP: number;
	classification: string;
	types: string[];
	resistant: string[];
	weaknesses: string[];
	evolutions: Pokemon[];
}
