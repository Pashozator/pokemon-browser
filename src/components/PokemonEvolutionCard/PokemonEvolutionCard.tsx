import React from 'react';
import './PokemonEvolutionCard.scss';
import Card from '../Card/Card';
import { Pokemon } from '../../types/pokemon';
import { Link } from 'react-router-dom';

interface PokemonEvolutionCardProps {
	evolution: Pokemon;
}

const PokemonEvolutionCard: React.FC<PokemonEvolutionCardProps> = ({ evolution }) => {
	return (
		<Link to={`/${evolution.name}`} className="pokemon-evolution">
			<button>
				<Card className="pokemon-evolution-card">
					<img src={evolution.image} alt={evolution.name}/>
					{evolution.name}
				</Card>
			</button>
		</Link>
	)
};

export default PokemonEvolutionCard;
