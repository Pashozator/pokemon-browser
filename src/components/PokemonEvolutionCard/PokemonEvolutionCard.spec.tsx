import React from 'react'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PokemonEvolutionCard from './PokemonEvolutionCard';
import { Pokemon } from '../../types/pokemon';

describe('PokemonEvolutionCard component', () => {
	it('should render', () => {
		const pokemon: Pokemon = {
			id: 'test',
			name: 'Pikachu',
			image: 'image'
		};
		const wrapper = shallow(<PokemonEvolutionCard evolution={pokemon}/>);

		expect(wrapper.exists()).toBeTruthy();
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
