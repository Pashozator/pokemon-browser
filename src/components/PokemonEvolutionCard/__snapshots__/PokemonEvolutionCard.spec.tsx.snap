// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonEvolutionCard component should render 1`] = `
<Link
  className="pokemon-evolution"
  to="/Pikachu"
>
  <button>
    <Card
      className="pokemon-evolution-card"
    >
      <img
        alt="Pikachu"
        src="image"
      />
      Pikachu
    </Card>
  </button>
</Link>
`;
